bson/src/main/org/bson/json/DateTimeFormatter.java

```diff
@@ -47,7 +47,7 @@
 
     private static FormatterImpl loadDateTimeFormatter(final String className) {
         try {
-            return (FormatterImpl) Class.forName(className).newInstance();
+            return (FormatterImpl) Class.forName(className).getDeclaredConstructor().newInstance();
         } catch (ClassNotFoundException e) {
```

Comment:
It is, but then the RuntimeException subclass of Exception is also caught, and there's no reason to catch RuntimeException.  Unfortunately there's no common superclass for all checked exceptions that can be caught in a single catch expression. (Java 7 has better syntax for this: http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html)

One could argue that it's ok to catch RuntimeException, but it's inconsistent from the rest of the driver to do so.